src/v2/style-guide/index.md

```diff
@@ -0,0 +1,1802 @@
+
+For the most part, we also avoid suggestions about JavaScript or HTML in general. We don't mind whether you use semicolons or trailing commas. We don't mind whether your HTML uses single-quotes or double-quotes for attribute values. Some exceptions will exist however, where we've found that a particular pattern is helpful in the context of Vue.
+
+> **Soon, we'll also provide tips for enforcement.** Sometimes you'll simply have to be disciplined, but wherever possible, we'll try to show you how to use ESLint and other automated processes to make enforcement simpler.
+
+Finally, we've split rules into four categories:
+
+
+
+## Rule Categories
+
+### Priority A: Essential
+
+These rules help prevent errors, so learn and abide by them at all costs. Exceptions may exist, but should be very rare and only be made by those with expert knowledge of both JavaScript and Vue.
+
+### Priority B: Strongly Recommended
+
+These rules have been found to vastly improve readability in most projects. If you violate them, your code will still run, but other developers may give you the evil eye [at conferences](https://github.com/vuejs/roadmap#events).
+
+[![Fry from Futurama glaring](https://i.imgur.com/PcOnLxn.jpg)](http://futurama.wikia.com/wiki/Template:FOX_Image)
+
+### Priority C: Recommended
+
+Where multiple, equally good options exist, an arbitrary choice can be made to ensure consistency. In these rules, we describe each acceptable option and suggest a default choice. That means you can feel free to make a different choice in your own codebase, as long as you're consistent and have a good reason. Please do have a good reason though! By adapting to the community standard, you will:
+
+1. train your brain to more easily parse most of the community code you encounter
+2. be able to copy and paste most community code examples without modification
+2. often find new hires are already accustomed to your preferred coding style, at least in regards to Vue
+
+### Priority D: Use with Caution
+
+Some features of Vue exist to accommodate rare edge cases or smoother migrations from a legacy code base. When overused however, they can make your code more difficult to maintain or even become a source of bugs. These rules shine a light on potentially risky features, describing when and why they should be avoided.
+
+[![goto from xkcd.com](https://imgs.xkcd.com/comics/goto.png)](https://xkcd.com/292/)
+
+
+
+## Priority A Rules: Essential (Error Prevention)
+
+
+
+### Multi-word component names <sup data-p="a">essential</sup>
+
+**Component names should always be multi-word, except for root `App` components.**
+
+This prevents conflicts with existing and future HTML elements, since all HTML elements are a single word.
```

Comment:
Here is a ref may be useful: [Valid custom element name in W3C Custom Elements Spec](http://w3c.github.io/webcomponents/spec/custom/#valid-custom-element-name)
The same concern for the name of W3C custom elements.

Reply:
Good idea!